Docker g++ compile errors

**Issue**

`compile_program.sh` seems to be unable to finish "Compiling main program" step for any ST file, including the blank program.

I get the following error:

> Compiling main program...
> /usr/bin/ld: /tmp/ccINpdYs.o: in function python_block_loader': python_loader.cpp:(.text+0x3a3): undefined reference to shm_open'
> /usr/bin/ld: python_loader.cpp:(.text+0x522): undefined reference to shm_open' /usr/bin/ld: python_loader.cpp:(.text+0x5a7): undefined reference to shm_unlink'
> /usr/bin/ld: python_loader.cpp:(.text+0x655): undefined reference to shm_unlink' /usr/bin/ld: python_loader.cpp:(.text+0x723): undefined reference to shm_unlink'
> collect2: error: ld returned 1 exit status
> Error compiling C files
> Compilation finished with errors!

**Build process**

Built image using

`docker build -t openplc:v3 .`

Ran using

`docker run -it --rm --privileged -p 8080:8080 openplc:v3`



I am unsure if this applies to all Linux systems or just the debian:bullseye used in the Docker setup.

--- a/webserver/scripts/compile_program.sh
+++ b/webserver/scripts/compile_program.sh
@@ -104,9 +104,9 @@ elif [ "$OPENPLC_PLATFORM" = "linux" ]; then
     ./glue_generator
     echo "Compiling main program..."
     if [ "$OPENPLC_DRIVER" = "sl_rp4" ]; then
-        g++ -std=gnu++11 *.cpp *.o -o openplc -I ./lib -pthread -fpermissive `pkg-config --cflags --libs libmodbus` -lsnap7 -lasiodnp3 -lasiopal -lopendnp3 -lopenpal -w $ETHERCAT_INC -DSL_RP4
+        g++ -std=gnu++11 *.cpp *.o -o openplc -I ./lib -pthread -fpermissive `pkg-config --cflags --libs libmodbus` -lsnap7 -lasiodnp3 -lasiopal -lopendnp3 -lopenpal -lrt -w $ETHERCAT_INC -DSL_RP4
     else
-        g++ -std=gnu++11 *.cpp *.o -o openplc -I ./lib -pthread -fpermissive `pkg-config --cflags --libs libmodbus` -lsnap7 -lasiodnp3 -lasiopal -lopendnp3 -lopenpal -w $ETHERCAT_INC 
+        g++ -std=gnu++11 *.cpp *.o -o openplc -I ./lib -pthread -fpermissive `pkg-config --cflags --libs libmodbus` -lsnap7 -lasiodnp3 -lasiopal -lopendnp3 -lopenpal -lrt -w $ETHERCAT_INC 
     fi
     if [ $? -ne 0 ]; then
         echo "Error compiling C files"

